src/cursor/cur_backup.c

```diff
@@ -207,22 +351,204 @@ __backup_log_append(WT_SESSION_IMPL *session, WT_CURSOR_BACKUP *cb, bool active)
     return (ret);
 }
 
+/*
+ * __backup_config --
+ *     Backup configuration.
+ */
+static int
+__backup_config(WT_SESSION_IMPL *session, WT_CURSOR_BACKUP *cb, const char *cfg[],
+  WT_CURSOR_BACKUP *othercb, bool *foundp, bool *log_only, bool *incr_only)
+{
+    WT_CONFIG targetconf;
+    WT_CONFIG_ITEM cval, k, v;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_ITEM(tmp);
+    WT_DECL_RET;
+    const char *uri;
+    bool incremental_config, is_dup, log_config, target_list;
+
+    conn = S2C(session);
+    *foundp = *incr_only = *log_only = false;
+    incremental_config = log_config = false;
+    is_dup = othercb != NULL;
+
+    /*
+     * Per-file offset incremental hot backup configurations take a starting checkpoint and optional
+     * maximum transfer size, and the subsequent duplicate cursors take a file object.
+     */
+    WT_RET_NOTFOUND_OK(__wt_config_gets(session, cfg, "incremental.force_stop", &cval));
+    if (cval.val) {
+        /*
+         * If we're force stopping incremental backup, set the flag. The resources involved in
+         * incremental backup will be released on cursor close and that is the only expected usage
+         * for this cursor.
+         */
+        if (is_dup)
+            WT_RET_MSG(session, EINVAL,
+              "Incremental force stop can only be specified on a primary backup cursor");
+        F_SET(cb, WT_CURBACKUP_FORCE_STOP);
+        return (0);
+    }
+    WT_RET_NOTFOUND_OK(__wt_config_gets(session, cfg, "incremental.enabled", &cval));
+    if (cval.val) {
+
+        if (!F_ISSET(conn, WT_CONN_INCR_BACKUP)) {
+            WT_RET(__wt_config_gets(session, cfg, "incremental.granularity", &cval));
+            /* XXX may not need cb->incr_granularity */
+            if (conn->ckpt_incr_granularity != 0)
+                WT_RET_MSG(session, EINVAL, "Cannot change the incremental backup granularity");
+            conn->ckpt_incr_granularity = cb->incr_granularity = (uint64_t)cval.val;
+        }
+        /* XXX Granularity can only be set once at the beginning */
+        F_SET(conn, WT_CONN_INCR_BACKUP);
+        incremental_config = true;
+    }
```

Comment:
Why can't we change the backup granularity? I don't see any point in it, but I don't see any reason it shouldn't work, either.

Reply:
It is discussed in the design. I recall it had to do with allocating elements for the block lists and what we've already persisted, etc. For now it seems like it is not needed. If it is needed later, we can add a new feature ticket to do so.

Reply:
I'm pretty sure there's no reason to disallow this, but I also don't see any reason anyone would change the setting, either.